sctp/src/packet.rs

```diff
@@ -155,30 +154,29 @@ impl Packet {
         writer.put_u16(self.destination_port);
         writer.put_u32(self.verification_tag);
 
+        // This is where the checksum will be written
+        let checksum_pos = writer.len();
+        writer.extend_from_slice(&[0, 0, 0, 0]);
+
         // Populate chunks
-        let mut raw = BytesMut::new();
         for c in &self.chunks {
-            let chunk_raw = c.marshal()?;
-            raw.extend(chunk_raw);
+            c.marshal_to(writer)?;
 
-            let padding_needed = get_padding_size(raw.len());
+            let padding_needed = get_padding_size(writer.len());
             if padding_needed != 0 {
-                raw.extend(vec![0u8; padding_needed]);
+                // padding needed if < 4 because we pad to 4
+                writer.extend_from_slice(&[0u8; 16][..padding_needed]);
```

Comment:
why do we need 16 bytes then? if max is 4

Reply:
actually it's a bit dangerous in a sense that `PADDING_MULTIPLE` may change and this will throw a runtime error. why can't we do `[0u8; padding_needed]`? because it's slower?

Reply:
> why can't we do `[0u8; padding_needed]`

Because rust doesn't allow dynamically sized arrays on the stack right?

> actually it's a bit dangerous in a sense that `PADDING_MULTIPLE` may change and this will throw a runtime error

This is a good point, if unlikely since the RFC for SCTP will probably never change in this regard. I can just use `PADDING_MULTIPLE` as the size though. since `padding_needed` is guaranteed to be < `PADDING_MULTIPLE`